### DISCUSSION TITLE
Intellij-Plugin: unsafe legacy renegotiation disabled

### DISCUSSION THREAD

**Discussion by DanielBednorz** [community/community#70608] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Hi there,

i have issues with copilot. When I'm logged in and type in something, getting any suggestions fails with this error:
"copilot is encountering temporary issues: E0570000:error:0A000152:SSL routines:final_renegotiation:unsafe legacy renegotiation disabled"

I'm behind a company VPN with a self signed certificate. 
I have followed this troubleshoot: https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-network-errors-for-github-copilot
It doesnt helped. I'm not a ssl-expert but for me it looks like there is no issue with the certificate. At least at this point.

I did a research what "unsafe legacy renegotiation disabled" means.  Here you go: https://github.com/openssl/openssl/issues/21296#issuecomment-1609397222

Does that mean it's an copilot issue?
There is a similar issue with VS2022 [here](https://github.com/orgs/community/discussions/69921#discussion-5729505). I just dont want to add a "me2-answer".

Some info about my system:
windows 10
System proxy / self signed certificate
IntelliJ IDEA 2023.2.2
Copilot Plugin 1.3.2.3479

**Comment by DanielBednorz** [comment]
Am I the only one having this issue?

**Reply by AshRC17** [reply]
I've seen it as well. I am working on the same problem with a Mac. Renegotiation more than likely has to do with the self signed cert and I found this article (https://jfrog.com/help/r/artifactory-why-am-i-observing-unsafe-legacy-renegotiation-disabled-during-npm-install/artifactory-why-am-i-observing-unsafe-legacy-renegotiation-disabled-during-npm-install) that pretty much describes the issue. I'm going to get a Mac that I can test out a couple of solutions on since I don't want to make changes to one of our developers Mac. I'll respond with what I find and if you find a work around please post here as well.

**Reply by AshRC17** [reply]
Found additional info here 
https://github.com/openssl/openssl/issues/18374

**Comment by Dineth-H** [comment]
The error message means that Copilot is unable to establish a secure connection to the GitHub Copilot service.

Here are some things you can try:

- Update Copilot to the latest version.
- Disable the Copilot plugin and then enable it again.
- Try restarting IntelliJ IDEA.
- Try connecting to a different network, such as your home Wi-Fi network, to see if Copilot is able to connect.
- Contact GitHub support for assistance.

Similar issues are there in VS Code, This suggests that the issue is likely caused by Copilot itself and If you are unable to resolve the issue on your own, you may need to wait for GitHub to release a fix.

In the meantime, you can still use Copilot by disabling the unsafe legacy renegotiation warning
